engine/src/main/java/org/terasology/rendering/dag/StateChange.java

```diff
@@ -22,4 +22,7 @@
     StateChange getDefaultInstance();
 
     RenderPipelineTask generateTask();
+
+    // TODO: investigate a better way?
+    boolean compare(StateChange stateChange);
```

Comment:
I agree that "equals" is a better name. However the reason of selecting `compare` rather than `equals` was completely trying to not spend time on `hashcode` and possible Java related problems. Any suggestions?